dali/imgcodec/decoders/nvjpeg2k.h

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_DECODERS_NVJPEG2K_H_
+#define DALI_IMGCODEC_DECODERS_NVJPEG2K_H_
+
+#include <nvjpeg.h>
+#include <memory>
+#include <vector>
+#include "dali/imgcodec/decoders/decoder_parallel_impl.h"
+#include "dali/imgcodec/decoders/nvjpeg/nvjpeg2k_helper.h"
+#include "dali/imgcodec/decoders/nvjpeg/nvjpeg_memory.h"
+#include "dali/core/dev_buffer.h"
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/core/cuda_event.h"
+
+namespace dali {
+namespace imgcodec {
+
+/**
+ * @brief Decoder interface for nvjpeg2k library
+ */
+class DLL_PUBLIC NvJpeg2000DecoderInstance : public BatchParallelDecoderImpl {
+ public:
+  NvJpeg2000DecoderInstance(int device_id, ThreadPool *tp);
+  ~NvJpeg2000DecoderInstance();
+
+  using BatchParallelDecoderImpl::CanDecode;
+  bool CanDecode(ImageSource *in, DecodeParams opts, const ROI &roi) {
+    // TODO(staniewzki): add support for roi and other data types
+    return !roi && (opts.dtype == DALI_UINT8);
```

Comment:
I see, that in the base signature, the `const ROI &roi = {}`. Is it possible, that this condition will ever pass?

Reply:
I don't exactly understand what you're asking about. For the default-constructed ROI, `.use_roi() == false`, so `!roi == true`.